cupyx/jit/_cuda_types.py

```diff
@@ -40,6 +40,14 @@ def __hash__(self):
         return hash(self.dtype)
 
 
+class PtrDiff(Scalar):
+    def __init__(self):
+        super().__init__('q')
+
+    def __str__(self):
+        return 'ptrdiff_t'
```

Comment:
How about using `Scalar(numpy.int64)` instead of `PtrDiff`?

Reply:
I think maybe we should use SFINAE and `is_signed`, `is_unsigned`, `is_floating_point` to implement `_floor_divide`, but static-casting to `long long` is indeed much simpler.😄